### DISCUSSION TITLE
Can't find a way to zoom all the way back when there are a lot of points

### DISCUSSION THREAD

**Discussion by ultrapoci** [patrykandpatrick/vico#384] [Ideas]
**Describe the bug**
I have a chart with 1000+ points. I would like to be able to scroll and zoom it. The problem is that when zoom is enabled, it's not possible to zoom all the way back and see the complete chart. If I disable zooming, the chart correctly shows all the points, but of course, I cannot zoom anymore. 

**Module**

- [x] `compose`
- [ ] `views`
- [ ] `compose-m2`
- [ ] `compose-m3`
- [ ] `core` (select if none of the above apply)

**To reproduce**
Create a chart with enough points (I'm using 1000) and enable zoom. 

**Expected behavior**
It should be possible to zoom a chart all the way back and see the complete chart, regardless of the number of points. 

**Android versions**
Vico v1.11.2
Android v11

**Comment by patrickmichalik** [comment]
Hello! At the moment, a chart’s zoom factor is limited to a constant interval, 0.1 to 10, with some exceptions. We’ll be looking into making this more flexible, enabling you to achieve the result you’re looking for. Thanks for the suggestion!

**Reply by ultrapoci** [reply]
I see, thank you for the quick reply!

**Reply by arijit4** [reply]
@patrickmichalik I was also finding a way to fit all points to the view at the same time. And user will also be able to zoom out and in.

Seeing this issue, I guess, a solve for this issue should work for me.

The behaviour I'm expecting:
1. The graph should show all the points at once at the beginning 
2. Users should be able to zoom in as their liking

Hope, you'll look into this soon.

**Reply by patrickmichalik** [reply]
The changes that we plan on implementing will make that possible, @arijit4. I’ll post a comment here once we’ve made these changes and released an update.

**Reply by JayantSharma777** [reply]
@patrickmichalik I am also implementing a line graph with x-axis being the time throughout the day, currently i am showing averaged hourly data to the user so that it fits my graph i.e only 24 points will be shown.
But what i want to achieve is that i should add all my data to the graph and not average it out and if the user wants to see the hourly data or half hourly data,he can just zoom in for better understanding of the data.
Now right now i can give an external dropdown in my activity to the user to choose if he wants to see 24 hour data ,or 1 hour data,30 minutes data and so on and I will do this by decreasing the spacing in the x-axis via AxisItemPlacer.
Please let me know if I can achieve this by any other means possible.
Here is how my graph looks right now:->
<img width="375" alt="image" src="https://github.com/patrykandpatrick/vico/assets/75779183/755b9d96-cdbc-47f7-ab26-3bba0d6957a4">
In this screen shot,I am showing the averaged data for every hour,but I feel it is ambiguous as it will not provide the right information
what i want is to set all my data points in the graph, and the user can zoom in and look more carefully at the data and the labels in x-axis should also change appropriately to show the correct time.
<img width="398" alt="image" src="https://github.com/patrykandpatrick/vico/assets/75779183/bd8ebcc4-6d71-43d4-9cbe-366e52bf80c2">
I want to show my graph like this in the begining, and the user can zoom in to look at the data more carefully.
Please let me know if there is any better approach i can implement using this library rather than the two i have mentioned above.
Thanks

**Reply by patrickmichalik** [reply]
My suggestion would be as follows, @JayantSharma777. First, lock the _x_ step, using the value from the first chart (i.e., the _x_ delta corresponding to one hour). You can do so via `getXStep`. At this point, you’ll be able to add intermediate entries without affecting the `HorizontalAxis` labels or the scale of the chart. Second, use `AxisItemPlacer.Horizontal` to implement the desired label behavior, as described under #424.

**Reply by JayantSharma777** [reply]
Thanks @patrickmichalik for this approach.I will implement it.Thanks

**Reply by patrickmichalik** [reply]
You’re welcome!

**Reply by mathklk** [reply]
Hi, is there any update on this? I have a similar problem and would like to zoom in/out more.

**Reply by patrickmichalik** [reply]
We’re currently finalizing Vico 2.0.0 Alpha 7, and we plan on introducing zoom customization in version 2.0.0 Alpha 8 or 2.0.0 Alpha 9, @mathklk. It’s the next thing we’ll be looking into.

**Comment by patrickmichalik** [comment]
[Vico 2.0.0 Alpha 9](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.9) addresses this. Cheers!

**Comment by Gowsky** [comment]
Hello. [Vico 2.0.0](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0) is now available.
